lib/compiler/src/engine/artifact.rs

```diff
@@ -319,7 +388,13 @@ impl Artifact {
             }),
         };
 
-        artifact.internal_register_frame_info();
+        // The error scenario could be improved here. We're deserializing frame info
+        // as part of registration, so that's returning a DeserializeError, but
+        // this function returns a CompileError that's then wrapped in a
+        // DeserializeError anyway.
+        artifact
+            .internal_register_frame_info()
+            .map_err(|e| CompileError::Validate(format!("{:?}", e)))?;
```

Comment:
Is this a Validation Error really? Maybe we can use other CompileError variant a bit more fit for a frame info registration error

Reply:
Well, not really. The problem is that we're now doing a bit of on-demand deserialization as part of the compile process, so compilation _can_ also return deserialization errors. I could change the return type to `DeserializeError`, but since `from_parts` is public, that would be a breaking change.